### PR TITLE
feat(mvt): add Arrow table tiling and migrate examples

### PR DESCRIPTION
**Pull request by ibgreen**
## Goals

Add a separate Arrow-native alternative to `TableTileSourceLoader` and migrate the table-tiler example and rendering integration to it, without changing the existing GeoJSON source.

## Changes

- Export lightweight `ArrowTableTileSourceLoader` metadata and public types from `@loaders.gl/mvt`; preload `ArrowTableTileSourceLoaderWithParser` and `ArrowTableVectorTileSource` through the explicit `@loaders.gl/mvt/arrow-table-tile-source-loader` subpath.
- Accept loaders.gl/Apache Arrow tables, promises, and URL/blob inputs through an injected Arrow-producing loader.
- Reuse existing clipping, wrapping, and simplification while keeping attributes in Arrow; gather contiguous source-row runs as zero-copy views.
- Write tile geometries directly to GeoArrow WKB buffers, retaining polygon holes and refreshing geometry metadata.
- Return Arrow consistently from all tile accessors; validate coordinate-system and indexing options, reject `indexMaxZoom > maxZoom`, and return null for tile requests beyond `maxZoom`.
- Move the existing native GeoArrow geometry reader into the lightweight GIS module, preserving its GeoArrow re-export.
- Add deterministic browser tests, a package-boundary bundle test, lazy-loading and zoom-boundary regressions, API documentation, release notes, and the direct Apache Arrow dependency.
- Migrate the standalone/website tile example and source guide to Arrow, update documentation tabs and the source overview, and preserve the legacy API documentation.
- Route geographic and tile-local Arrow tiles through deck.gl's binary geometry rendering path, honoring the selected primary geometry column.
- Make a single explicitly configured parser authoritative so deck.gl's registered JSON loader cannot intercept GeoJSON input.
- Fix the example's stable loader identity, current panels API, layer ID, selection label, and no-emit TypeScript configuration; correct a shared style value exposed by strict example type checking.
- Emit unknown local CRS using each metadata format's convention: omitted GeoArrow CRS, explicit null GeoParquet CRS.

## Scope

Two-dimensional, in-memory tiling of WGS84 longitude/latitude geometry. This does not add worker execution, reprojection, out-of-core querying, or encoded MVT output.

## Validation

- `yarn` and `yarn lint fix`
- `yarn build` and `yarn build-workers`
- `yarn test-node`: 458 passed
- `yarn test-headless`: 5,096 passed
- `yarn test-audit`: zero violations
- Standalone tiles example: `yarn build` (TypeScript and Vite)
- Website: `yarn build`
- Built-example Chromium smoke test with a synthetic polygon and a stubbed basemap: visible geometry, no console/page errors
- Verified published ESM metadata preload and CommonJS implementation subpath
- Final revision `cd9d97a71` passed all required CI checks, including merged coverage, Node 22/24/26, both Chromium shards, affected hermetic slow tests, build, and website build.

### CODE REVIEW

File: modules/mvt/src/index.ts [hunk 1]
```diff
@@ -53,5 +53,14 @@ export type {MVTSourceLoaderOptions} from './mvt-source-loader';
 export {TableTileSourceLoader, TableVectorTileSource} from './table-tile-source-loader';
 export type {TableTileSourceLoaderOptions} from './table-tile-source-loader';
 
+export {
+  ArrowTableTileSourceLoader,
+  ArrowTableVectorTileSource
+} from './arrow-table-tile-source-loader';
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/mvt/src/index.ts]: Fixed in cd9d97a71. The package root now exports metadata from its own loader-types file (plus type-only source/options exports). A static preload imports the explicit @loaders.gl/mvt/arrow-table-tile-source-loader subpath, which exposes ArrowTableTileSourceLoaderWithParser and the runtime class. Shared defaults/types live in a neutral file. Updated documentation, added a bundle-boundary regression and async-load integration coverage, and verified ESM preload, CommonJS subpath, website/example builds, and the production example browser smoke test. Full validation: 458 Node tests and 5,096 Chromium tests pass.

File: modules/mvt/src/arrow-table-tile-source-loader.ts [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+function validateTilingOptions(options: ArrowTableTileSourceLoaderOptions): void {
+  const tableOptions = {...TableTileSourceLoader.defaultOptions.table, ...options.table};
+  for (const name of ['maxZoom', 'indexMaxZoom'] as const) {
+    if (
+      !Number.isInteger(tableOptions[name]) ||
+      tableOptions[name] < 0 ||
+      tableOptions[name] > 24
+    ) {
+      throw new Error(`${name} must be an integer in the 0-24 range`);
+    }
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/mvt/src/arrow-table-tile-source-loader.ts:259]: Fixed in cd9d97a71. Validation now rejects indexMaxZoom > maxZoom before indexing, and getTileSync explicitly rejects requests above the declared maxZoom. Added the reported invalid-options regression plus both pre-indexed and on-demand indexing boundary tests. The options and out-of-range behavior are documented. Full validation: 458 Node tests and 5,096 Chromium tests pass.
